### DISCUSSION TITLE
Case sensitive routing doesn't work

### DISCUSSION THREAD

**Discussion by andriiradkevych** [freshframework/fresh#2327] [General]
My seo team , gave me task to make routing working , without relation to capital letters , is it possible ? What I mean `/Regiser` and `/register` should work . But now `/Register` redirects me on `404` page . 
Also in addition , I see that it works on `localhost`, but it doesn't when it is deployed on domain. Can't find out the reason . Seems like localhost browser just transform , but domain doesn't . Any ideas ?

In prodcution , it just started on docker

```
FROM denoland/deno:1.39.1

ARG GIT_REVISION
ENV DENO_DEPLOYMENT_ID=${GIT_REVISION}

WORKDIR /app

COPY . .
RUN deno task cache
EXPOSE 3000
CMD ["task", "preview"]
```

also , here is my `_middleware`:
```
import { MiddlewareHandlerContext } from '$fresh/server.ts';

import { isMobile as isMobileDevice } from 'utils/device.ts';

export function handler(req: Request, ctx: MiddlewareHandlerContext) {
  const userAgent = req.headers.get('user-agent');

  if (userAgent) {
    ctx.state.isMobile = isMobileDevice(userAgent);
  }
  return ctx.next();
}
``` 

 maybe it is possible to fix here without pain and impact ?

**Comment by marvinhagemeister** [comment]
> My seo team , gave me task to make routing working , without relation to capital letters

On an unrelated note: It's the first time I'm hearing that URL casing has an effect on SEO. I'm curious to know if there is some truth behind it.

Back to topic: You can write a middleware that redirects all upper/mixed-cased routes to the lowercased ones.
